src/Compress/Gz.php

```diff
@@ -152,19 +173,19 @@ public function compress($content)
      *
      * @param  string $content
      * @return string
-     * @throws Exception\RuntimeException if unable to open archive or error during decompression
+     * @throws Exception\RuntimeException If unable to open archive or error during decompression.
      */
     public function decompress($content)
     {
         $archive = $this->getArchive();
         $mode    = $this->getMode();
 
         //check if there are null byte characters before doing a file_exists check
-        if (false === strpos($content, "\0") && file_exists($content)) {
+        if (null !== $content && false === strpos($content, "\0") && file_exists($content)) {
```

Comment:
as unchanged lines cannot be commented i throw it here...
given `$content` is `null` it would raise an error further below:
```
} elseif ($mode === 'deflate') {
    $compressed = gzinflate($content);
} else {
    $compressed = gzuncompress($content);
}
```

this seems to be uncovered in tests (`DecompressTest` is testing this with bz2)

Reply:
Indeed, I added an explicit check for `null` values and extended tests in 73cd4bd

Reply:
See https://github.com/laminas/laminas-filter/pull/35#pullrequestreview-817519564 for follow up discussion on null values.